WikiExtension/src/timeSeriesService.js

```diff
@@ -0,0 +1,59 @@
+import fetch from "node-fetch";
+
+const formatDateToYYYMMDD = (date) =>
+    `${date.getFullYear()}${("0" + (date.getMonth() + 1)).slice(-2)}${("0" + date.getDate()).slice(-2)}`;
+
+const formatYYYYMMDDToDate = (yyyymmdd) =>
+    new Date(yyyymmdd.substring(0, 4), yyyymmdd.substring(4, 6) - 1, yyyymmdd.substring(6, 8));
+
+/**
+ * Fetches page views information on a given wikipedia title. The response will only include stat since its creation
+ * if the startDate comes before the article was created.
+ *
+ * @param {string} title of a wikipedia article
+ * @param {Date} startTimestamp in a format of YYYYMMDD
+ * @param {Date} endDate in a format of YYYYMMDD
+ * @param {string} aggregateType of either monthly or daily
+ * @return {map} of an item of list with properties project, article, granularity(aggregationType), timestamp, access, agent, views
+ */
+const fetchPageViews = async (title, startDate, endDate, aggregateType) => {
+    const formattedStartDate = formatDateToYYYMMDD(startDate);
+    const formattedEndDate = formatDateToYYYMMDD(endDate);
+    const response = await fetch(
+        `https://wikimedia.org/api/rest_v1/metrics/pageviews/per-article/en.wikipedia/all-access/all-agents/${title}/${aggregateType}/${formattedStartDate}/${formattedEndDate}`
+    );
+    return await response.json();
+};
+
+/**
+ * TODO: Subject to change based on what the graph needs
+ * @return formated fetchedPageViews as a list of lists that only contain date object and count
+ */
+const formatPageViews = (fetchedPageViews) => {
+    return Array.from(fetchedPageViews.items).map((pageViewObject) => {
+        return [formatYYYYMMDDToDate(pageViewObject.timestamp), pageViewObject.views];
+    });
+};
+
+/**
+ * Gets view counts on a given article within a given time range.
+ *
+ * @param {string} title of a wikipedia article
+ * @param {Date} startDate
+ * @param {Date} endDate
+ * @param {AggregateType} aggregateType an enum that represents string "monthly" or "daily"
+ * @returns aggregated list of data of views per day or per month
+ */
+const getPageViews = async (title, startDate, endDate, aggregateType) => {
+    try {
+        let result = await fetchPageViews(title, startDate, endDate, aggregateType);
+        let formattedResult = formatPageViews(result);
+        return formattedResult;
+    } catch (err) {
+        console.error(
```

Comment:
Should we also return null or error here? Console.log is useful for us, but for the user we could display a message "Couldn't load graph" instead of nothing when this function returns error

Reply:
The messaging user will probably be the front end's job. I will pass err out to make it easier to display messages. Added here: [a03dca7](https://github.com/sukritsangvong/wp-browser-extension/pull/5/commits/a03dca732fbe594b6ad1c7b8c4fc5f100fa5b5dc)

Reply:
Agree, the null here is all we need